src/raw.rs

```diff
@@ -292,6 +303,47 @@ pub fn get_string() -> io::Result<String> {
     }
 }
 
+/// Retrieves a list of file paths from the `CF_HDROP` format.
+///
+/// # Pre-conditions:
+///
+/// * [open()](fn.open.html) has been called.
+pub fn get_file_list() -> io::Result<Vec<PathBuf>> {
+    unsafe {
+        let clipboard_data = GetClipboardData(formats::CF_HDROP).if_null_get_last_error()?;
+        let _data_ptr = GlobalLock(clipboard_data).if_null_get_last_error()?;
+        let gen_err_unlock = || -> io::Error {
+            GlobalUnlock(clipboard_data);
+            io::ErrorKind::Other.into()
+        };
+        let num_files = DragQueryFileW(clipboard_data as HDROP, std::u32::MAX, ptr::null_mut(), 0);
+        let file_names: io::Result<Vec<PathBuf>> = (0..num_files)

```

Comment:
Done, although I don't generally prefer this style since you have to manually set the `Vec` capacity (which the Iterator will do automatically when collecting otherwise), plus the `Vec` has to be mutable and pushing is manual.